### PR TITLE
DAOS-9165 control: force format if unknown fs sig

### PR DESCRIPTION
**Pull request by tanabarr**
If --reformat is specified, proceed to format even if fs on device is
unrecognised or cannot be otherwise determined. this should avoid
situations where a PMEM device cannot be formatted because of unusual 
ilesystem signatures.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Review by mjmac — COMMENTED** [review]
This looks good to me, except that now `parseFsType()` can never return an error, so the return signature should probably be changed. And in the switch statement, there should be a default case that returns an error, just to be defensive.

**Comment by tanabarr** [comment]
whilst the solution looks good, I've been able to replicate the issue demonstrated in ticket DAOS-9165 even after scrambling the data on device with dd:
```
tanabarr@wolf-157:~/projects/daos_m> sudo su -c "head -c 128k </dev/urandom >/dev/pmem0"
tanabarr@wolf-157:~/projects/daos_m> sudo file -s /dev/pmem0
/dev/pmem0: data
ftanabarr@wolf-157:~/projects/daos_m> sudo su -c "dd if=/dev/urandom of=/dev/pmem0 bs=4k count=1k"
1024+0 records in
1024+0 records out
4194304 bytes (4.2 MB, 4.0 MiB) copied, 0.0289705 s, 145 MB/s
tanabarr@wolf-157:~/projects/daos_m> sudo file -s /dev/pmem0
/dev/pmem0: data
```
`/dev/pmem0: data` is parseable and results in successful format with --reformat even on master

**Review by kjacque — APPROVED** [review]

File: src/control/server/storage/scm/provider.go [hunk 1]
```diff
@@ -472,6 +473,12 @@ func (p *Provider) CheckFormat(req storage.ScmFormatRequest) (*storage.ScmFormat
 		res.Mountable = true
 	case fsTypeNone:
 		res.Formatted = false
+	case fsTypeUnknown:
+		// formatted but not mountable
+		p.log.Debugf("unexpected format of output from 'file -s %s'", req.Dcpm.Device)
+	default:
```
Comment (knard-intel) [inline src/control/server/storage/scm/provider.go:479]: From what I understand, the default case should never happen. Thus, an error should probably be returned.

Comment (tanabarr) [reply src/control/server/storage/scm/provider.go:479]: the default case will happen for any fs that is recognised but not one of the supported types e.g. there is a test case for this situation with ext2: https://github.com/daos-stack/daos/pull/7604/files/1a406044f09afb993b8c199b77ecbb6fc0769950#diff-c6c1b87e2db9b0eb1ce91d071551c308f10e18ec1583da175197c1f2ee1893d8R1058

Comment (knard-intel) [reply src/control/server/storage/scm/provider.go:479]: Thanks for the explanation, I misunderstood the default usage.
So everything seems OK for me :)

**Review by knard-intel — APPROVED** [review]
Fine for me :)

**Pull request merged**